sale_order_report_without_price/report/sale_report_templates.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template
+        id="report_saleorder_document_without_price"
+        inherit_id="sale.report_saleorder_document"
+        primary="True"
+        priority="99"
+    >
+        <xpath expr="//th[@name='th_priceunit']" position="replace" />
+        <xpath expr="//th[@name='th_discount']" position="replace" />
+        <xpath expr="//th[@name='th_taxes']" position="replace" />
+        <xpath expr="//th[@name='th_subtotal']" position="replace" />
+
+        <xpath expr="//td[@name='td_priceunit']" position="replace" />
+        <xpath expr="//td[count(@name) = 0]" position="replace" />
```

Comment:
You should get discount 'td' with [@t-if='display_discount'] for being more specific